utils/babeltrace_thapi.in

```diff
@@ -424,17 +442,4 @@ $options[:'backend-names'] = $options[:backends].map { |name_level| name_level.s
 #
 # Setup Logger
 LOGGER = Logger.new($stdout)
-
-$thapi_graph = { 'tally' => ['source.ctf.fs', 'filter.utils.muxer', 'filter.intervals.interval',
-                             'filter.btx_aggreg.aggreg', 'sink.btx_tally.tally'],
-                 'timeline' => ['source.ctf.fs', 'filter.utils.muxer', 'filter.intervals.interval',
-                                'sink.btx_timeline.timeline'],
-                 'trace' => ['source.ctf.fs', 'filter.utils.muxer', 'sink.text.rubypretty'],
-                 'trace_live' => ['source.ctf.lttng_live', 'filter.utils.muxer', 'sink.text.rubypretty'],
-                 'to_interval' => ['source.ctf.fs', 'filter.utils.muxer', 'filter.intervals.interval',
-                                   'filter.btx_stripper.stripper', 'sink.ctf.fs'],
-                 'to_aggreg' => ['source.ctf.fs', 'filter.utils.muxer', 'filter.intervals.interval',
-                                 'filter.btx_aggreg.aggreg', 'filter.btx_stripper.stripper', 'sink.ctf.fs'] }
-
-command = 'trace_live' if command == 'trace' && $options[:live]
-create_and_run_graph(command, ARGV)
+create_and_run_graph(command, bt_graphs[command], ARGV)
```

Comment:
```suggestion
create_and_run_graph(command, bt_graph(command), ARGV)
```
Also needed if suggestion above is applied.

Reply:
Graph cached, so built only once. Same memory footprint as a static global variable